ksqldb-common/src/main/java/io/confluent/ksql/function/types/ParamTypes.java

```diff
@@ -61,7 +83,11 @@ public static boolean areCompatible(
       final SqlMap sqlType = (SqlMap) argumentSqlType;
       final MapType mapType = (MapType) declared;
       return areCompatible(SqlArgument.of(sqlType.getKeyType()), mapType.key(), allowCast)
-          && areCompatible(SqlArgument.of(sqlType.getValueType()), mapType.value(), allowCast);
+          && areCompatible(
+          SqlArgument.of(sqlType.getValueType()),
```

Comment:
With lambdas above we use `areCompatible(SqlType, ParamType)` which wraps to a `SqlArgument` and then passes down to `areCompatible(SqlArgument, ParamType, AllowCasts)`. Seems like we should either use the wrapper `areCompatible` for arrays and maps too or get rid of the extra class and set everything to `SqlArgument` here. Thoughts? @stevenpyzhang

Reply:
Let's just change the function to take SqlArgument also.